lib/test_cmr/load_data.rb

```diff
@@ -57,14 +58,19 @@ def load_data
 
     def wait_for_cmr
       cmr_up = false
-      until cmr_up
-        begin
-          cmr_up = yield
-        rescue Faraday::ConnectionFailed
-          puts 'CMR is still starting, please wait...'
-          sleep 5
+      Timeout::timeout(Rails.configuration.cmr_startup_timeout.to_i) do
+        until cmr_up
+          begin
+            cmr_up = yield
+          rescue Faraday::ConnectionFailed
+            puts 'CMR is still starting, please wait...'
+            sleep 5
+          end
         end
       end
+    rescue Timeout::Error => e
+      puts('Got timeout waiting for CMR, existing task!')
```

Comment:
should this probably be Logger.error instead of puts

Reply:
Or maybe both.

Reply:
Rake tasks don't have logger setup. Existing task use 'puts', which come into Bamboo log file.